prime-router/src/test/kotlin/fhirengine/engine/FhirConverterTests.kt

```diff
@@ -279,4 +280,54 @@ class FhirConverterTests {
             engine.getTransformerFromSchema("src/test/resources/fhir_sender_transforms/sample_schema")
         ).isNotNull()
     }
+
+    @Test
```

Comment:
If I'm reading this correctly, the test verifies that we made a report for the first bundle, and then started processing the second bundle before any queue messages were sent. I think that makes sense, it just took me a minute to follow, I think largely because the `RuntimeException` confused me. Maybe a comment to help explain that the `RuntimeException` is being used to return control at a specific point in `doWork`? Optional, might just be me, and either way, it made sense after looking at the function a bit more closely.

Reply:
Adding a comment makes sense to me!  But yes, you are correct the exception is arbitrary and just makes us exit early to confirm we don't send messages.